library/connect.c

```diff
@@ -305,10 +354,21 @@ int ziti_dial(ziti_connection conn, const char *service, ziti_conn_cb conn_cb, z
     req->conn = conn;
     req->cb = conn_cb;
 
+    if (dial_opts != NULL) {
+        // clone dial_opts to survive the async request
+        req->dial_opts = clone_ziti_dial_opts(dial_opts);
+
+        // override connection timeout if set in dial_opts
+        if (dial_opts->connect_timeout_seconds != 0) {
```

Comment:
technically connect_timeout_seconds is an int and 'could' be null... should this be > 0? (same comment for the other location - line 843)